Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp

```diff
@@ -1626,89 +1626,46 @@ ThrowCompletionOr<TemporalTimeZone> parse_temporal_time_zone_string(GlobalObject
 {
     auto& vm = global_object.vm();
 
+    auto optional_string_view_to_optional_string = [](Optional<StringView> const& s) {
+        return s.has_value() ? String { *s } : Optional<String> {};
+    };
```

Comment:
This is a bit ugly (but fine for now), making `Optional<T>` be explicitly construct-able from `Optional<U>` as long as `T` is construct-able from `U` sounds like a nice future improvement, e.g.
```c++
template<typename U = T>
ALWAYS_INLINE explicit Optional(Optional<U> const& other) requires(IsConstructible<T, U const&>)
    : m_has_value(other.m_has_value)
{
    if (other.has_value()) {
        new (&m_storage) T(other.value());
    }
}

template<typename U = T>
ALWAYS_INLINE explicit Optional(Optional<U>&& other) requires(IsConstructible<T, U&&>)
    : m_has_value(other.m_has_value)
{
    if (other.has_value()) {
        new (&m_storage) T(other.release_value());
    }
}
```
but @alimpfard can probably make it nicer

Reply:
That already looks pretty much perfect (as perfect as this stuff gets anyway)
I'd just suggest putting the `requires` immediately after the `template<...>` to avoid instantiating the method when it _definitely_ isn't needed.

Reply:
+1 for having 'native' support for this!